packages/docs/src/components/DocSiteNavigation.tsx

```diff
@@ -123,57 +132,117 @@ const DocSiteNavigation = ({ location }: DocSiteNavProps) => {
    * Checks sub nav items to see if any are currently selected
    */
   const isSubNavItemSelected = (subNavItems) => {
-    return subNavItems.some((navItem) => isItemSelected(navItem.url));
+    return subNavItems.some((navItem) => navItem.selected);
   };
 
   /**
-   * Updating a name to remove kebab case & get rid of numeric ordering
+   * Updating a name to remove kebab case
    */
   const formatNavItemLabel = (name: string): string => {
-    let newName = name.replace(/-/g, ' ');
-    newName = removePositioning(newName);
-    return newName;
+    return name.replace(/-/g, ' ');
   };
 
-  const formatNavItemData = ({ name, relativePath }: NavItem) => {
+  /**
+   * transforms from graphql structure to structure for <VerticalNav> `items` prop
+   */
+  const formatNavItemData = ({ childMdx, relativePath }: NavItem) => {
+    const { frontmatter } = childMdx;
+    const name = frontmatter.title;
     const url = makePageUrl(relativePath);
     return {
       label: formatNavItemLabel(name),
       url,
-      id: url,
+      id: relativePath,
       selected: isItemSelected(url),
+      order: frontmatter.order || 0,
     };
   };
 
-  const formatNavData = (dataList: GraphQlNavItem[]): VerticalNavItemProps[] => {
-    const retVal: VerticalNavItemProps[] = [];
+  /**
+   * Need to nest level2 items that have subnav items under the relevant level1 items
+   */
+  const nestSubNavs = (dataList): VerticalNavItemProps[] => {
+    const level1Items: { string?: VerticalNavItemProps } = {};
+    const level2Items = [];
+
+    // sort level1 vs level2 items
     dataList.forEach((dataItem) => {
```

Comment:
This `dataItem` is being interpreted by TypeScript as `any` because the type isn't defined in the function signature above. However, I think the `GraphQlNavItem[]` type that `formatNavData` expects—and what this function expects by extension—is not accurate. If you were to apply that type to `dataList`, it'd start throwing errors. Each of those `GraphQlNavItemNode` objects should have an `edges` and `fieldValue` property.

Reply:
I added some more specific type checking in the new functions